File: server/schemas/ui_test_suite.py
"""UI 任务配置 Schema"""
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class UITestSuiteCreate(BaseModel):
    """创建 UI 任务配置"""
    project_id: int
    name: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None
    case_ids: list[int] = []
    environment_id: Optional[int] = None
    failure_strategy: str = "continue"
    browser: str = "chrome"
    viewport_width: int = 1280
    viewport_height: int = 720
    tags: list[str] = []


class UITestSuiteUpdate(BaseModel):
    """更新 UI 任务配置"""
    name: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = None
    case_ids: Optional[list[int]] = None
    environment_id: Optional[int] = None
    failure_strategy: Optional[str] = None
    browser: Optional[str] = None
    viewport_width: Optional[int] = None
    viewport_height: Optional[int] = None
    tags: Optional[list[str]] = None


class UITestSuiteOut(BaseModel):
    """UI 任务配置输出"""
    id: int
    project_id: int
    project_name: Optional[str] = None
    name: str
    description: Optional[str] = None
    case_ids: list[int] = []
    case_count: int = 0
    environment_id: Optional[int] = None
    environment_name: Optional[str] = None
    failure_strategy: str
    browser: str
    viewport_width: int
    viewport_height: int
    tags: list[str] = []
    creator_id: Optional[int] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: frontend/src/api/uiTestSuite.ts
import request from '@/utils/request'

/** UI 任务配置 */
export interface UITestSuite {
  id: number
  project_id: number
  project_name?: string
  name: string
  description?: string
  case_ids: number[]
  case_count: number
  environment_id?: number
  environment_name?: string
  failure_strategy: string
  browser: string
  viewport_width: number
  viewport_height: number
  tags: string[]
  created_at: string
  updated_at?: string
}

/** 创建 UI 任务配置参数 */
export interface UITestSuiteCreate {
  project_id: number
  name: string
  description?: string
  case_ids: number[]
  environment_id?: number
  failure_strategy: string
  browser: string
  viewport_width: number
  viewport_height: number
  tags?: string[]
}

/** 更新 UI 任务配置参数 */
export interface UITestSuiteUpdate {
  name?: string
  description?: string
  case_ids?: number[]
  environment_id?: number
  failure_strategy?: string
  browser?: string
  viewport_width?: number
  viewport_height?: number
  tags?: string[]
}

/** 获取 UI 任务配置列表 */
export function getUITestSuites(params?: {
  project_id?: number
  keyword?: string
  page?: number
  page_size?: number
}) {
  return request.get('/ui-test-suites', { params })
}

/** 获取 UI 任务配置详情 */
export function getUITestSuite(suiteId: number) {
  return request.get<UITestSuite>(`/ui-test-suites/${suiteId}`)
}

/** 创建 UI 任务配置 */
export function createUITestSuite(data: UITestSuiteCreate) {
  return request.post<UITestSuite>('/ui-test-suites', data)
}

/** 更新 UI 任务配置 */
export function updateUITestSuite(suiteId: number, data: UITestSuiteUpdate) {
  return request.put<UITestSuite>(`/ui-test-suites/${suiteId}`, data)
}

/** 删除 UI 任务配置 */
export function deleteUITestSuite(suiteId: number) {
  return request.delete(`/ui-test-suites/${suiteId}`)
}

/** 批量删除 UI 任务配置 */
export function batchDeleteUITestSuites(suiteIds: number[]) {
  return request.post('/ui-test-suites/batch-delete', { suite_ids: suiteIds })
}

/** 执行 UI 任务配置 */
export function runUITestSuite(suiteId: number) {
  return request.post<{ id: number; name: string }>(`/ui-test-suites/${suiteId}/run`)
}
